hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/paging/SizeAwarePage.java

```diff
@@ -50,7 +50,7 @@ public SizeAwarePage(Iterator<T> wrapped, PageContext pageContext, HasTotalSize
     @Override
     public long getTotalSize() {
         if (totalSize == HasTotalSize.NOT_DEPLETED && !hasNext()) {
-            totalSize = hasTotalSize.getTotalSize();
+            totalSize = hasTotalSize == null ? HasTotalSize.NOT_DEPLETED : hasTotalSize.getTotalSize();
```

Comment:
Technically, setting `totalSize` to `NOT_DEPLETED` here is not correct, right? Because in this block we know we've depleted the iterator, it's just that `hasTotalSize` has already been nulled out (which I guess can only happen when you somehow call `close()` and `hasNext()` still returns true).